src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -323,6 +323,12 @@ function VisEditor($rootScope, $scope, $route, timefilter, AppState, $window, kb
       if (id) {
         notify.info('Saved Visualization "' + savedVis.title + '"');
         if ($scope.isAddToDashMode()) {
+          // We need to manually set the last url for this app so navigating back to visualize will open the saved
+          // visualization, not the unsaved one.
+          const savedVisualizationUrl =
+            kbnUrl.eval(`${kbnBaseUrl}#${VisualizeConstants.EDIT_PATH}/{{id}}`, { id: savedVis.id });
+          chrome.trackSubUrlForApp('kibana:visualize', savedVisualizationUrl);
```

Comment:
This is only going to update the url in the nav, it's not going to add/edit browser history like this should be doing.  Instead of telling the chrome to track a new suburl, how about we tell the browser to replace the current item in history before moving on to the next url? This way the chrome will see the navigation, update the app link, and the back button will work too.